cips/cip-21.md

```diff
@@ -84,7 +84,42 @@ Test cases will need to ensure that a user may not forge a incorrect signer, nor
 
 ## Reference Implementation
 
-TBC
+The go implementation makes the following modifications to the `Blob` proto:
+
+```diff
+message BlobProto {
+  bytes namespace_id = 1;
+  bytes data = 2;
+  uint32 share_version = 3;
+  uint32 namespace_version = 4;
++ // Signer is sdk.AccAddress that paid for this blob. This field is optional
```

Comment:
[question] can this feature be used with feegrant? If yes, and we have:

1. celestiaA agrees to pay gas for all message for celestiaB 
2. celestiaB submits an authored blob and uses gas from celestiaA. 
3. Whose address goes in the signer field?

If it's celestiaB then should we stop using the term "that paid for this blob" and instead use something like: 

```go
// Signer is sdk.AccAddress that submitted this blob.
```

Reply:
good catch!

Reply:
After reading through this CIP again recently, I'm a little confused how this would handle blobs submitted with feegrant as described in this thread.